### DISCUSSION TITLE
NXP missing board.h file in fsl_flexspi_nor_boot.h

### DISCUSSION THREAD

**Discussion by DerekSnell** [zephyrproject-rtos/zephyr#59280] [Q&A] [answered]
This question comes from @Nadav from https://discord.com/channels/720317445772017664/883445396284772413/1118144085744681101

Hello everyone!
After updating the zephyr version to 3.1.0, mcuboot wont build.
it fails with the error: 
```
In file included from /home/nadav/zephyrproject/zephyr/soc/arm/nxp_imx/rt/soc_rt11xx.c:20:
/home/nadav/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1176/xip/fsl_flexspi_nor_boot.h:14:10: fatal error: board.h: No such file or directory
```

Before updating to the latest release, everything worked perfectly fine.
I came across this: https://github.com/zephyrproject-rtos/zephyr/discussions/45559 is the solution provided by @Derek.Snell is the one i need to take into consideration?
I am working on a custom board based on mimxrt1172.

Thanks in advanced!

**Comment by DerekSnell** [comment, marked as answer]
Hi Nadav,

Zephyr is about to release v3.4, so I am currently working from the tag `v3.4.0-rc3`.  I was able to build the MCUboot application for the board `mimxrt1170_evk_cm7` using the command below:
```
west build -b mimxrt1170_evk_cm7 ../bootloader/mcuboot/boot/zephyr
```

Modifying this example, I found the `board.h` reference in [fsl_flexspi_nor_boot.h](https://github.com/zephyrproject-rtos/hal_nxp/blob/904830e8f684a9fd573751a1cdecde877ec49242/mcux/mcux-sdk/devices/MIMXRT1176/xip/fsl_flexspi_nor_boot.h#L14) is not included.  I confirmed this by renaming this board.h file, and the app still builds.  The reason it is not included is the macro `BOARD_FLASH_SIZE` is defined.

Search the Zephyr tree for the `BOARD_FLASH_SIZE` macro, it is defined in the board's [CMakeLists.txt](https://github.com/zephyrproject-rtos/zephyr/blob/785d9bdc6709ed47616aec2ef09fabcea4e06ddf/boards/arm/mimxrt1170_evk/CMakeLists.txt#L22) file, using the line below:
```
zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE*1024)
```

You mentioned you are building with a custom board.  My guess is your CMake file is not defining this `BOARD_FLASH_SIZE` macro.

Let us know what you find, best regards

**Reply by nadavdw** [reply]
Hi Derek ,
Thank you for the post.

The suggested solution solved my problem.  

Adding the line:
`zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE*1024)`
To my custom board CMakeLists.txt file indeed solved my issue.

Thanks again, BR.
